### PR TITLE
Tweaks to adding a newly created domain to the cache

### PR DESCRIPTION
**Pull request by ross**
Follow on research to https://github.com/octodns/octodns-dnsmadeeasy/pull/18, intended to be merged there, if anything. 

/cc See https://github.com/octodns/octodns-dnsmadeeasy/pull/18 for more details

### CODE REVIEW

File: octodns_dnsmadeeasy/__init__.py [hunk 1]
```diff
@@ -102,10 +102,15 @@ def domain(self, name):
         return self._request('GET', path).json()
 
     def domain_create(self, name):
-        self._request('POST', '/', data={'name': name})
+        resp = self._request('POST', '/', data={'name': name})
+        _id = resp.json()['data'][0]['id']
```
Comment (yzguy) [inline octodns_dnsmadeeasy/__init__.py:106]: It's just going to be `resp.json()['id']`, the response doesn't have a list of data like the `domains` resp has

Comment (ross) [reply octodns_dnsmadeeasy/__init__.py:106]: Ah. I was probably getting a mock that was intended for a domains call.

**Comment by yzguy** [comment]
Added some bits

Unsure why this is failing, if I remove the one above it it works fine. Do the context handlers not reset the mock?

```python
        # Non-existent zone doesn't populate anything
        with requests_mock() as mock:
            with open('tests/fixtures/dnsmadeeasy-no-domains.json') as fh:
                mock.get(f'{base}/', text=fh.read())

            zone = Zone('unit.tests.', [])
            provider.populate(zone)
            self.assertEqual(0, len(zone.records))

        # No diffs == no changes
        with requests_mock() as mock:
            with open('tests/fixtures/dnsmadeeasy-domains.json') as fh:
                mock.get(f'{base}/', text=fh.read())
            with open('tests/fixtures/dnsmadeeasy-records.json') as fh:
                mock.get(f'{base}/123123/records', text=fh.read())

                zone = Zone('unit.tests.', [])
                provider.populate(zone)
>               self.assertEqual(14, len(zone.records))
E               AssertionError: 14 != 0
```

**Comment by ross** [comment]
> Unsure why this is failing, if I remove the one above it it works fine. Do the context managers not reset the mock? Will need to come back to this later this weekend.

Maybe it's the domains and/or records cache on the provider instance?

**Comment by yzguy** [comment]
Closing this in favor of #21

**Pull request closed without merging**